tools/azsdk-cli/Azure.Sdk.Tools.Cli/Services/APIView/APIViewAuthenticationService.cs

```diff
@@ -0,0 +1,164 @@
+                return null;
+            }
+
+            TokenRequestContext tokenRequest = new([scope]);
+            AccessToken tokenResponse = await credential.GetTokenAsync(tokenRequest);
+
+            _logger.LogInformation("Successfully obtained Azure token with scope {Scope}", scope);
+            return tokenResponse.Token;
+        }
+        catch (Exception ex)
+        {
+            _logger.LogWarning(ex, "Failed to get Azure token with scope {Scope}", APIViewConfiguration.ApiViewScopes.GetValueOrDefault(environment, "unknown"));
+            return null;
+        }
+    }
+
+    public async Task ConfigureAuthenticationAsync(HttpClient httpClient, string environment = "production")
+    {
+        string? token = await GetAuthenticationTokenAsync(environment);
+        if (!string.IsNullOrEmpty(token))
+        {
+            httpClient.DefaultRequestHeaders.Authorization =
+                new AuthenticationHeaderValue("Bearer", token);
+
+            httpClient.DefaultRequestHeaders.Add("X-API-Key", token);
+        }
+
+        httpClient.DefaultRequestHeaders.Add("User-Agent", APIViewConfiguration.UserAgent);
+        httpClient.DefaultRequestHeaders.Add("Accept", "application/json");
+
+        _logger.LogDebug("Request headers configured - Authorization: {HasAuth}, User-Agent: {UserAgent}",
+            !string.IsNullOrEmpty(token) ? "Bearer [TOKEN]" : "None", APIViewConfiguration.UserAgent);
+    }
+
+    public string GetAuthenticationGuidance()
+    {
+        return @"Authentication is required to access APIView data. Please authenticate using Azure credentials:
+
+**Azure CLI**:
+   - Run: az login
+   - Ensure you have access to APIView resources
+
+The authentication service will automatically try Azure CLI";
+    }
+
+    public async Task<AuthenticationStatus> CheckAuthenticationStatusAsync(string? environmentOption = null)
+    {
+        string environment = environmentOption ?? "production";
+        string baseUrl = APIViewConfiguration.BaseUrlEndpoints[environment];
+        string? token = await GetAuthenticationTokenAsync(environment);
+        
+        bool hasToken = !string.IsNullOrEmpty(token);
+        bool isAuthenticationWorking = false;
+        string? authenticationError = null;
+
+        if (hasToken)
+        {
+            try
+            {
+                using var httpClient = new HttpClient();
+                await ConfigureAuthenticationAsync(httpClient, environment);
+                
+                string testUrl = $"{baseUrl}/api/authtest/status";
+                using var response = await httpClient.GetAsync(testUrl);
+                
+                if (response.IsSuccessStatusCode)
+                {
+                    string responseContent = await response.Content.ReadAsStringAsync();
+                    isAuthenticationWorking = responseContent.Contains("\"IsAuthenticated\":true", StringComparison.OrdinalIgnoreCase);
```

Comment:
Why do we need this? Shouldn't end point return 401 in case of authentication failure and 200 only when auth works fine? I would prefer not to do this string parsing to decide the state of a request whenever possible.